docs/sources/articles/dockerfile_best-practices.md

```diff
@@ -291,32 +291,28 @@ auto-extraction capability, you should always use `COPY`.
 
 ### [`ENTRYPOINT`](https://docs.docker.com/reference/builder/#entrypoint)
 
-The best use for `ENTRYPOINT` is as a helper script. Using `ENTRYPOINT` for
-other tasks can make your code harder to understand. For example,
+The best use for `ENTRYPOINT` is to set the image's main command, allowing that
+image to be run as though it was that command (and then to use `CMD` the default
+options).
 
-....docker run -it official-image bash
+Let's start with an example of an image for the command line tool `s3cmd`:
 
-is much easier to understand than
+    ENTRYPOINT ["s3cmd"]
+    CMD ["--help"]
 
-....docker run -it --entrypoint bash official-image -i
+Now the image can be run like this:
 
-This is especially true for new Docker users, who might naturally assume the
-above command will work fine. In cases where an image uses `ENTRYPOINT` for
-anything other than just a wrapper script, the command will fail and the
-beginning user will then be forced to learn about `ENTRYPOINT` and
-`--entrypoint`.
+    $ docker run s3cmd ls s3://mybucket
```

Comment:
It might be good to show running the image twice, once with the "ls" parameters and once without them. That way it shows how/when `--help` will be used.  Also, perhaps show the output (or explain what the output would be) in each case.